source/extensions/filters/http/ext_proc/mutation_utils.cc

```diff
@@ -161,20 +161,66 @@ absl::Status MutationUtils::applyHeaderMutations(const HeaderMutation& mutation,
       rejected_mutations.inc();
       return absl::InvalidArgumentError("Invalid character in set_headers mutation.");
     }
+
+    auto check_op = CheckOperation::SET;
+    auto should_append = false;
     const LowerCaseString header_name(sh.header().key());
-    const bool append = PROTOBUF_GET_WRAPPED_OR_DEFAULT(sh, append, false);
-    const auto check_op = (append && !headers.get(header_name).empty()) ? CheckOperation::APPEND
-                                                                        : CheckOperation::SET;
-    auto check_result = checker.check(check_op, header_name, header_value);
-    if (replacing_message && header_name == Http::Headers::get().Method) {
-      // Special handling to allow changing ":method" when the
-      // CONTINUE_AND_REPLACE option is selected, to stay compatible.
-      check_result = CheckResult::OK;
+    const auto header_exists = !headers.get(header_name).empty();
+
+    if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.ext_proc_legacy_append")) {
+      // Legacy append behavior
+      should_append = PROTOBUF_GET_WRAPPED_OR_DEFAULT(sh, append, false);
+      check_op = (should_append && header_exists) ? CheckOperation::APPEND : CheckOperation::SET;
+    } else {
+      if (sh.has_append()) {
```

Comment:
Thanks for reviewing this! Let me address your questions in order:

1. We have implemented a check to prevent users from specifying both `append` and `append_action`. Additionally, a behavior change notice is included to inform users that if neither option is explicitly set, the default behavior of `append_action` will apply, which is to append the header instead of replacing it.
2. Allowing both options to be set simultaneously is not permitted. If users attempt to do so, the system will throw an `InvalidArgumentError`.
3. That’s correct! If one option is set and the other remains default, the non-default setting takes precedence. However, we would issue a deprecation warning, as `append` is being marked as deprecated.
4. While we aim to maintain compatibility, the design cannot be fully backward compatible. This is because there’s no way to distinguish in the proto whether `append` was left unspecified or explicitly set to `false`. As a result, a minor behavior change is necessary. This is also noted as part of the changelog.